packages/amplify-graphql-relational-transformer/src/has-many/has-many-directive-transformer-factory.ts

```diff
@@ -3,15 +3,30 @@ import { HasManyDirectiveConfiguration } from '../types';
 import { DataSourceBasedDirectiveTransformer } from '../data-source-based-directive-transformer';
 import { HasManyDirectiveDDBFieldsTransformer } from './has-many-directive-ddb-fields-transformer';
 import { HasManyDirectiveSQLTransformer } from './has-many-directive-sql-transformer';
+import { HasManyDirectiveDDBReferencesTransformer } from './has-many-directive-ddb-references-transformer';
 
 export const getHasManyDirectiveTransformer = (
   dbType: ModelDataSourceStrategyDbType,
+  config: HasManyDirectiveConfiguration,
+  // eslint-disable-next-line consistent-return
 ): DataSourceBasedDirectiveTransformer<HasManyDirectiveConfiguration> => {
   switch (dbType) {
     case 'MYSQL':
     case 'POSTGRES':
       return new HasManyDirectiveSQLTransformer(dbType);
     case 'DYNAMODB':
+      // If references are passed to the directive, we'll use the references relational
+      // modeling approach.
+      if (config.references?.length >= 1) {
```

Comment:
This is improved in https://github.com/aws-amplify/amplify-category-api/pull/2373